src/main/java/com/emc/pravega/perf/PravegaWriterWorker.java

```diff
@@ -32,82 +29,41 @@
-        this.eventsPerSec = eventsPerSec;
-        this.secondsToRun = secondsToRun;
-        this.StartTime = start;
-        this.stats = stats;
-        this.isRandomKey = isRandomKey;
-        this.messageSize = messageSize;
-        this.totalEvents = totalEvents;
+                        PerfStats stats, String streamName, long totalEvents,
+                        ClientFactory factory) {
+
+        super(sensorId, eventsPerSec, secondsToRun,
+                isRandomKey, messageSize, start,
+                stats, streamName, totalEvents);
+
         this.producer = factory.createEventWriter(streamName,
                 new UTF8StringSerializer(),
                 EventWriterConfig.builder().build());
     }
 
-    /**
-     * This function will be executed in a loop and time behavior is measured.
-     *
-     * @return A function which takes String key and data and returns a future object.
-     */
-    public CompletableFuture writeData(String key, String data) throws TxnFailedException, IllegalStateException {
+    @Override
+    public CompletableFuture writeData(String key, String data) throws IllegalStateException {
         return producer.writeEvent(key, data);
     }
 
     @Override
-    public Void call() throws TxnFailedException, InterruptedException, ExecutionException, IllegalStateException {
-        CompletableFuture retFuture = null;
-
-        do {
-            final Instant loopStartTime = Instant.now();
-            for (int i = 0; (i < eventsPerSec) && (eventCount.incrementAndGet() <= totalEvents) &&
-                    (Duration.between(loopStartTime, Instant.now()).getSeconds() < 1); i++) {
-
-                // Construct event payload
-                String val = System.currentTimeMillis() + ", " + producerId + ", " + (int) (Math.random() * 200);
-                String payload = String.format("%-" + messageSize + "s", val);
-                String key;
-                if (isRandomKey) {
-                    key = Integer.toString(producerId + new Random().nextInt());
-                } else {
-                    key = Integer.toString(producerId);
-                }
-
-                final Instant startTime = Instant.now();
-                retFuture = writeData(key, payload);
-                // event ingestion
-                retFuture = stats.recordTime(retFuture, startTime, payload.length());
-            }
-
-            long timeSpent = Duration.between(loopStartTime, Instant.now()).toMillis();
-            if (timeSpent < 1000) {
-                Thread.sleep(1000 - timeSpent);
-            }
-        } while ((Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun) &&
-                (eventCount.get() < totalEvents));
-
+    public void flush() {
         producer.flush();
-        // producer.close();
+    }
 
-        //Wait for the last packet to get acked
-        retFuture.get();
+    @Override
+    public long eventCountIncrementAndGet() {
```

Comment:
Why is this a public method?

Reply:
this is overriding the abstract method fromt the abstract class WriterWorker

Reply:
ok, i have removed them now